### ISSUE TITLE
hyperkube fails to run

### ISSUE THREAD

**Issue by brosner** [kubernetes/kubernetes#14838]
I am testing out the v1.1.0 tag and built a hyperkube binary. However, running it with no arguments is producing:

```
_output/local/go/bin/hyperkube 
W0930 17:57:47.051620     714 server.go:176] Failed to start in resource-only container "/kube-proxy": mkdir /sys/fs/cgroup/perf_event/kube-proxy: read-only file system
W0930 17:57:47.051962     714 server.go:185] Neither --kubeconfig nor --master was specified.  Using default API client.  This might not work.
W0930 17:57:47.053839     714 iptables.go:140] Error checking iptables version, assuming version at least 1.4.11: exec: "iptables": executable file not found in $PATH
E0930 17:57:47.054589     714 server.go:331] Not trying iptables proxy: can't get Node "507aa6445e16": Get http://localhost:8080/api/v1/nodes/507aa6445e16: dial tcp 127.0.0.1:8080: connection refused
F0930 17:57:47.055092     714 server.go:246] Unable to create proxier: failed to initialize iptables: error creating chain "KUBE-PORTALS-CONTAINER": exec: "iptables": executable file not found in $PATH: 
```

I would have expected a usage message. None of the servers are working correctly. After doing some more digging I found that when adding the proxy server to `hk` is causing it to essentially boot up breaking hyperkube.

See 1c25c2cd9954d2985d115e3dc876918222aff7db

**Comment by cjcullen** [comment]
@gmarek

**Comment by gmarek** [comment]
I don't know why me. IIUC hyperkube needs an argument naming the server to run (e.g. kubelet, proxy, apiserver etc.). I don't know why we're not printing usage. I also don't know who's the author - @thockin @bgrant0607 @mikedanese do you know?

**Comment by brosner** [comment]
The change in 1c25c2c caused the breakage because `kubeproxy.NewProxyServerDefault` is being called while adding servers to hyperkube. In my development environment I moved the call to `kubeproxy.NewProxyServerDefault` under the `hks.Run` function and it works as expected.

**Comment by mikedanese** [comment]
@brosner is iptables on your PATH?

**Comment by gmarek** [comment]
Ugh... we do have a fix for this for 2 days already: #14660, but we also had trouble in our testing infrastructure and merging was slow:/

**Comment by brosner** [comment]
@mikedanese yes, I know it says it cannot locate `iptables` but that isn't the issue. Look like @gmarek found a PR that this is duplicate of. I am happy to close this in favor of that.
